File: app/src/main/java/adapter/CrewAdapter.java
package adapter;

import android.content.Context;
import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.ProgressBar;
import android.widget.TextView;

import androidx.fragment.app.FragmentActivity;
import androidx.recyclerview.widget.RecyclerView;

import com.google.firebase.analytics.FirebaseAnalytics;
import com.squareup.picasso.Callback;
import com.squareup.picasso.Picasso;

import java.util.List;

import br.com.icaro.filme.R;
import domain.CrewItem;
import pessoa.activity.PersonActivity;
import utils.Constantes;
import utils.UtilsApp;

/**
 * Created by icaro on 22/02/17.
 */
public class CrewAdapter extends RecyclerView.Adapter<CrewAdapter.CrewViewHolder> {
    private Context context;
    private List<CrewItem> crews;

    public CrewAdapter(FragmentActivity activity, List<CrewItem> crews) {
        context = activity;
        this.crews = crews;
    }


    @Override
    public CrewViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(context).inflate(R.layout.scroll_crews, parent, false);
        return new CrewViewHolder(view);
    }

    @Override
    public void onBindViewHolder(final CrewViewHolder holder, final int position) {
        holder.progressBarCrew.setVisibility(View.VISIBLE);
        final CrewItem crew = crews.get(position);

        if (crew.getName() != null && crew.getJob() != null) {
            holder.textCrewJob.setText(crew.getJob());
            holder.textCrewNome.setText(crew.getName());
            Picasso.get()
                    .load(UtilsApp.INSTANCE.getBaseUrlImagem(UtilsApp.INSTANCE.getTamanhoDaImagem(context, 4)) + crew.getProfilePath())
                    .placeholder(R.drawable.person)
                    .into(holder.imgPagerCrews, new Callback() {
                        @Override
                        public void onSuccess() {
                            holder.progressBarCrew.setVisibility(View.GONE);
                        }

                        @Override
                        public void onError(Exception e) {
                            holder.progressBarCrew.setVisibility(View.GONE);
                        }

                    });
        } else {
            holder.textCrewJob.setVisibility(View.GONE);
            holder.textCrewNome.setVisibility(View.GONE);
            holder.progressBarCrew.setVisibility(View.GONE);
            holder.imgPagerCrews.setVisibility(View.GONE);
        }

        holder.imgPagerCrews.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View view) {
                Intent intent = new Intent(context, PersonActivity.class);
                intent.putExtra(Constantes.PERSON_ID, crew.getId());
                intent.putExtra(Constantes.NOME_PERSON, crew.getName());
                context.startActivity(intent);

                Bundle bundle = new Bundle();
                bundle.putString(FirebaseAnalytics.Event.SELECT_CONTENT, PersonActivity.class.getName());
                bundle.putInt(FirebaseAnalytics.Param.ITEM_ID, crew.getId());
                bundle.putString(FirebaseAnalytics.Param.ITEM_NAME, crew.getName());
                FirebaseAnalytics.getInstance(context).logEvent(FirebaseAnalytics.Event.SELECT_CONTENT, bundle);
            }
        });
    }

    @Override
    public int getItemCount() {
        return crews.size();
    }

    class CrewViewHolder extends RecyclerView.ViewHolder {
        private TextView textCrewJob, textCrewNome;
        private ImageView imgPagerCrews;
        private ProgressBar progressBarCrew;

        CrewViewHolder(View itemView) {
            super(itemView);

             textCrewJob = itemView.findViewById(R.id.textCrewJob);
             textCrewNome = itemView.findViewById(R.id.textCrewNome);
             imgPagerCrews = itemView.findViewById(R.id.imgPagerCrews);
             progressBarCrew = itemView.findViewById(R.id.progressBarCrews);
        }
    }
}

File: app/src/main/java/adapter/CrewAdapter.kt
package adapter

import Layout
import android.content.Context
import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.ImageView
import android.widget.ProgressBar
import android.widget.TextView
import androidx.recyclerview.widget.RecyclerView.*
import br.com.icaro.filme.R.*
import domain.CrewItem
import domain.ViewType
import pessoa.activity.PersonActivity
import pessoaspopulares.adapter.ViewTypeDelegateAdapter
import utils.Constant
import utils.gone
import utils.setPicassoWithCacheAndHolder
import utils.visible

/**
 * Created by icaro on 22/02/17.
 */
class CrewAdapter(val context: Context, horizontal: Boolean) : ViewTypeDelegateAdapter {
	val layout: Int =
		if (horizontal) Layout.works_layout_horizontal else Layout.works_layout_vertical

	override fun onCreateViewHolder(parent: ViewGroup) = CrewViewHolder(parent)
	override fun onBindViewHolder(holder: ViewHolder, item: ViewType?, context: Context?) {
		(holder as CrewViewHolder).bind(item!!)
	}

	inner class CrewViewHolder(parent: ViewGroup) :
		ViewHolder(LayoutInflater.from(context).inflate(layout, parent, false)) {
		private val textCrewJob: TextView = itemView.findViewById(id.person_work)
		private val textCrewNome: TextView = itemView.findViewById(id.name_work)
		private val img: ImageView = itemView.findViewById(id.img_work)
		private val progressBarCrew: ProgressBar = itemView.findViewById(id.progress_work)
		fun bind(crewItem: ViewType) = with(itemView) {
			progressBarCrew.visible()
			val (_, _, name, profilePath, id, _, job) = crewItem as CrewItem
			textCrewJob.text = job
			textCrewNome.text = name
			img.setPicassoWithCacheAndHolder(stillPath = profilePath,
				patten = 2,
				img_erro = drawable.person,
				holder = drawable.person,
				sucesso = { progressBarCrew.gone() },
				error = { progressBarCrew.gone() })


			setOnClickListener {
				context.startActivity(Intent(context, PersonActivity::class.java).apply {
					putExtra(Constant.PERSON_ID, id)
					putExtra(Constant.NOME_PERSON, name)
				})
			}
		}
	}
}